infra/src/main/scala/ch/epfl/scala/services/storage/sql/SqlRepo.scala

```diff
@@ -143,13 +146,20 @@ class SqlRepo(conf: DatabaseConfig, xa: doobie.Transactor[IO])
 
   // to use only when inserting one element or updating one element
   // when expecting a row to be modified
-  private def run[A](i: A => doobie.Update0, v: A): Future[A] =
-    i(v).run.transact(xa).unsafeToFuture.flatMap {
-      case 1 => Future.successful(v)
-      case code =>
-        Future.failed(new Exception(s"Failed to insert $v (code: $code)"))
+  private def strictRun(
+      update: doobie.Update0
+  )(expectedRows: Int): Future[Unit] =
```

Comment:
maybe a default value to 1!

This run was meant to be used only by insert, since when inserting a row, we expect to have only 1 modified line, but by mistake I used it once when doing insertOrUpdate and it was failing when we don't update anything. 
Did you change it because of something similar ?